adapters/base/src/main/java/org/apache/cassandra/sidecar/adapters/base/exception/OperationUnavailableException.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.adapters.base.exception;
+
+/**
+ * Indicates that the operation is unavailable because a requirement for the operation has not been met
+ */
+public class OperationUnavailableException extends RuntimeException
+{
+    private final OperationType operationType;
+
+    public OperationUnavailableException(OperationType operationType)
+    {
+        super(operationType.message);
+        this.operationType = operationType;
+    }
+
+    public OperationType operationType()
+    {
+        return operationType;
+    }
+
+    /**
+     * The requirement for the operation
+     */
+    public enum OperationType
+    {
+        /**
+         * Gossip is required for the operation, but gossip has been disabled
+         */
+        GOSSIP_DISABLED("Gossip is required for the operation but it is disabled"),
```

Comment:
nit: `GOSSIP_DISABLED` is the reason, not the operation type. Maybe rename the enum to `Cause` or so, or rename the variant name to `GOSSIP`, which is a type. A follow up question is what if an operation is unavailable due to multiple causes, how is the exception going to represent it? It is a bit inflexible. Probably consider constructing the exception with just `string` error message (and `operationType()` is not used), but declare a few enum for common exceptions, and for simplicity. For example, 

```java
public class OperationUnavailableException extends RuntimeException
{
    public static final OperationUnavailableException GOSSIP_DISABLED = new OperationUnavailableException("Gossip is required for the operation but it is disabled");

    public OperationUnavailableException(String errorMessage)
    {
        super(errorMessage);
    }
}
```

Reply:
I think the suggestion looks cleaner. Applying the suggestion!